commit a37154211ac658fa70ba9c70b3c0bfe930a38cd6
Author: karthick111 <karthickrko61@gmail.com>
Date:   Tue Jul 26 13:41:30 2016 +0200

    DisplayDevice: Backwards compatibility with old EGL
    
    Original implementation by OndraOrg of XDA:
    for more information: http://forum.xda-developers.com/showpost.php?p=34469675&postcount=19
    
    Change-Id: Ifa2f6ebd3991cbd208dd33a6d7e9a4ff2d026f85

diff --git a/services/surfaceflinger/Android.mk b/services/surfaceflinger/Android.mk
index 6a90083..1147b05 100644
--- a/services/surfaceflinger/Android.mk
+++ b/services/surfaceflinger/Android.mk
@@ -57,6 +57,10 @@ ifeq ($(TARGET_DISABLE_TRIPLE_BUFFERING),true)
     LOCAL_CFLAGS += -DTARGET_DISABLE_TRIPLE_BUFFERING
 endif
 
+ifeq ($(BOARD_EGL_NEEDS_FNW),true)
+    LOCAL_CFLAGS += -DEGL_NEEDS_FNW
+endif
+
 ifeq ($(TARGET_FORCE_HWC_FOR_VIRTUAL_DISPLAYS),true)
     LOCAL_CFLAGS += -DFORCE_HWC_COPY_FOR_VIRTUAL_DISPLAYS
 endif
diff --git a/services/surfaceflinger/DisplayDevice.cpp b/services/surfaceflinger/DisplayDevice.cpp
index b83149b..a2166d7 100644
--- a/services/surfaceflinger/DisplayDevice.cpp
+++ b/services/surfaceflinger/DisplayDevice.cpp
@@ -27,6 +27,10 @@
 #include <ui/DisplayInfo.h>
 #include <ui/PixelFormat.h>
 
+#ifdef EGL_NEEDS_FNW
+#include <ui/FramebufferNativeWindow.h>
+#endif
+
 #include <gui/Surface.h>
 
 #include <hardware/gralloc.h>
@@ -90,7 +94,11 @@ DisplayDevice::DisplayDevice(
 {
     Surface* surface;
     mNativeWindow = surface = new Surface(producer, false);
+#ifndef EGL_NEEDS_FNW
     ANativeWindow* const window = mNativeWindow.get();
+#else
+    ANativeWindow* const window = new FramebufferNativeWindow();
+#endif
     char property[PROPERTY_VALUE_MAX];
 
     /*
